src/ui/osx/TogglDesktop/Swift/Extensions/NSColor+VisibleColor.swift

```diff
@@ -8,25 +8,40 @@
 
 import Foundation
 
-@objc extension NSColor {
+extension NSColor {
 
     @objc func visibleColor() -> NSColor {
         guard let window = NSApplication.shared.keyWindow else { return self }
         let darkMode = window.isDarkMode
         let bottomThreshold: CGFloat = 75.0 / 255.0
         let topThreshold: CGFloat = 180.0 / 255.0
 
+        // Get correct RGB from current color space
+        guard let component = getRGBComponents() else { return self }
+        let red = component.red
+        let green = component.green
+        let blue = component.blue
+
         // If darkmode and color is black-based
         // or light mode and color is white-based
-        if (darkMode && (redComponent + blueComponent + greenComponent) <= 3.0 * bottomThreshold) ||
-            (!darkMode && (redComponent + blueComponent + greenComponent) >= 3.0 * topThreshold) {
+        if (darkMode && (red + blue + green) <= 3.0 * bottomThreshold) ||
+            (!darkMode && (red + blue + green) >= 3.0 * topThreshold) {
 
             // Revert color to make it visible
             let red = 1.0 - redComponent
             let green = 1.0 - greenComponent
             let blue = 1.0 - blueComponent
```

Comment:
@NghiaTranUIT I've just realized that I missed an error here. `redComponent`, `greenComponent`, `blueComponent` should not be used here anymore as we are using `red`, `green`, `blue` when extracting components.

Reply:
You're right. Good catcher 👍 I will update it now